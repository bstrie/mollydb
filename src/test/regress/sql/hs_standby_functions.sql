--
-- Hot Standby tests
--
-- hs_standby_functions.sql
--

-- should fail
select txid_current();

select length(txid_current_snapshot()::text) >= 4;

select mdb_start_backup('should fail');
select mdb_switch_xlog();
select mdb_stop_backup();

-- should return no rows
select * from mdb_prepared_xacts;

-- just the startup process
select locktype, virtualxid, virtualtransaction, mode, granted
from mdb_locks where virtualxid = '1/1';

-- suicide is painless
select mdb_cancel_backend(mdb_backend_pid());
