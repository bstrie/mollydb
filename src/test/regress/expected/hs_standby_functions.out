--
-- Hot Standby tests
--
-- hs_standby_functions.sql
--
-- should fail
select txid_current();
ERROR:  cannot execute txid_current() during recovery
select length(txid_current_snapshot()::text) >= 4;
 ?column? 
----------
 t
(1 row)

select mdb_start_backup('should fail');
ERROR:  recovery is in progress
HINT:  WAL control functions cannot be executed during recovery.
select mdb_switch_xlog();
ERROR:  recovery is in progress
HINT:  WAL control functions cannot be executed during recovery.
select mdb_stop_backup();
ERROR:  recovery is in progress
HINT:  WAL control functions cannot be executed during recovery.
-- should return no rows
select * from mdb_prepared_xacts;
 transaction | gid | prepared | owner | database 
-------------+-----+----------+-------+----------
(0 rows)

-- just the startup process
select locktype, virtualxid, virtualtransaction, mode, granted
from mdb_locks where virtualxid = '1/1';
  locktype  | virtualxid | virtualtransaction |     mode      | granted 
------------+------------+--------------------+---------------+---------
 virtualxid | 1/1        | 1/0                | ExclusiveLock | t
(1 row)

-- suicide is painless
select mdb_cancel_backend(mdb_backend_pid());
ERROR:  canceling statement due to user request
